File: src/main/java/ukalus/graph/NodeStack.java
package ukalus.graph;

import java.util.LinkedList;

public class NodeStack implements NodeCollection {

  private LinkedList<Node> storage = new LinkedList<Node>();

  public void add(Node e) {
    storage.addFirst(e);
  }

  public Node get() {
    return storage.peek();
  }

  public void remove() {
    storage.poll();
  }

  public boolean isEmpty() {
    return storage.isEmpty();
  }

  public String toString() {
    return storage.toString();
  }
}

File: src/main/java/ukalus/graph/NodeStack.kt
package ukalus.graph

import java.util.LinkedList

class NodeStack : NodeCollection {

    private val storage = LinkedList<Node>()

    override fun add(e: Node) = storage.addFirst(e)

    override fun get(): Node? = storage.peek()

    override fun remove() {
        storage.poll()
    }

    override val isEmpty
        get() = storage.isEmpty()

    override fun toString() = storage.toString()
}
